proxy/http/Http1ClientSession.h

```diff
@@ -181,7 +189,12 @@ class Http1ClientSession : public ProxyClientSession
 
   MIOBuffer *read_buffer;
   IOBufferReader *sm_reader;
-  C_Read_State read_state;
+
+  /*
+   * Volatile should not be necessary, but there appears to be a bug in the 4.9 rhel gcc
+   * compiler that was using an old version of read_state to make decisions in really_destroy
+   */
+  volatile C_Read_State read_state;
```

Comment:
If there are threaded dependencies on this, then we should force an atomic read or a memory acquire/release. I think the volatile is too subtle to be depended on.